src/lib.rs

```diff
@@ -132,6 +156,78 @@ impl<T: Transport + Sized> TapSigner<T> {
         }
         read_response
     }
+
+    pub fn certs_check(&mut self, nonce: Vec<u8>) -> Result<String, Error> {
+        let card_nonce = self.card_nonce.clone();
+
+        let certs_cmd = CertsCommand::default();
+        let certs_response: CertsResponse = self.transport.transmit(certs_cmd)?;
+
+        let check_cmd = CheckCommand::new(nonce.clone());
+        let check_response: Result<CheckResponse, Error> = self.transport.transmit(check_cmd);
+        if let Ok(response) = &check_response {
+            self.card_nonce = response.card_nonce.clone();
+        }
+
+        assert!(verify_tapsigner_signature(
+            &self.pubkey, 
+            check_response.unwrap().auth_sig.clone(), 
+            card_nonce, 
+            nonce.clone(), 
+            &self.secp
+        ).is_ok());
+
+        let mut pubkey = self.pubkey.clone();
+        for sig in &certs_response.cert_chain() {
+            // BIP-137: https://github.com/bitcoin/bips/blob/master/bip-0137.mediawiki
+            let subtract_by = match sig[0] {
+                27..=30 => 27, // P2PKH uncompressed
+                31..=34 => 31, // P2PKH compressed
+                35..=38 => 35, // Segwit P2SH
+                39..=42 => 39, // Segwit Bech32
+                _ => panic!("Unrecognized BIP-137 address")
+            };
+            let rec_id = RecoveryId::from_i32((sig[0] as i32) - subtract_by).unwrap();
+            let (_, sig) = sig.split_at(1);
+            let rec_sig = RecoverableSignature::from_compact(sig, rec_id).unwrap();
+            let md = Message::from_hashed_data::<sha256::Hash>(pubkey.serialize().as_slice());            
+            pubkey = self.secp.recover_ecdsa(&md, &rec_sig).unwrap();
+        }
+
+        // todo - as a constant?
```

Comment:
Here's an idea for how to use const values for this. But for this PR I don't mind leaving it as is so we can think about it some more. 

```rust
const PROD_FACTORY_ROOT_KEY:&str = "03028a0e89e70d0ec0d932053a89ab1da7d9182bdc6d2f03e706ee99517d05d9e1";
const TEST_FACTORY_ROOT_KEY:&str = "027722ef208e681bac05f1b4b3cc478d6bf353ac9a09ff0c843430138f65c27bab";

pub enum FactoryRootKey {
    Prod(PublicKey),
    Test(PublicKey),
}

impl TryFrom<PublicKey> for FactoryRootKey {
    type Error = Error;

    fn try_from(pubkey: PublicKey) -> Result<Self, Self::Error> {
        if pubkey.serialize().to_hex().eq(PROD_FACTORY_ROOT_KEY) {
            Ok(FactoryRootKey::Prod(pubkey))
        } else if pubkey.serialize().to_hex().eq(TEST_FACTORY_ROOT_KEY) {
            Ok(FactoryRootKey::Test(pubkey))
        } else {
            Err(Error::CertCheck(
                "Root cert is not from Coinkite. Card is counterfeit.".to_string(),
            ))
        }
    }
}

impl Debug for FactoryRootKey {
    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
        match &self {
            FactoryRootKey::Prod(pk) => {
                write!(f, "FactoryRootKey::Prod({:?})", pk)
            }
            FactoryRootKey::Test(pk) => {
                write!(f, "FactoryRootKey::Test({:?})", pk)
            }
        }
    }
}
```